app/src/main/java/com/osfans/trime/util/WeakHashSet.kt

```diff
@@ -12,7 +12,7 @@ class WeakHashSet<T> : MutableSet<T> {
 
     override fun iterator(): MutableIterator<T> = core.keys.iterator()
 
-    override fun add(element: T) = core.put(element, PlaceHolder) != null
+    override fun add(element: T) = core.put(element, PlaceHolder) == null
```

Comment:
@ShellWen Now actually java.util.Collections has static method newSetFromMap that allow to create a WeakHashSet from WeakHashMap, see https://github.com/fcitx5-android/fcitx5-android/blob/7152901a8edcc030b419fe21e8d8abf6aecb1169/app/src/main/java/org/fcitx/fcitx5/android/utils/Utils.kt#L243 for reference.